modules/modules.go

```diff
@@ -32,6 +35,15 @@ type Manager struct {
 // UserInvisibleModule is an option for `RegisterModule` that marks module not visible to user. Modules are user visible by default.
 func UserInvisibleModule(m *module) {
 	m.userVisible = false
+
+	// by default invisible modules should not be targetable.
+	m.targetable = false
```

Comment:
With this approach `TargettableModule` and `UserInvisibleModule` must be used in the correct order (`UserInvisibleModule` first and `TargettableModule` second).

Instead of `TargettableModule` we can perhaps have `UserInvisibleButTargettableModule` (better name needed) and use it instead of `UserInvisibleModule`, not in addition to it.

Reply:
actually that's what i did at first, then i changed my mind to do it this way because of the complicated name, I used `UserInvisibleTargetableModule`. But you're right about the order having to be correct, I haven't considered that, so I'll change it back to `UserInvisibleTargetableModule`

Reply:
done in ea759669